src/browser/browser.ts

```diff
@@ -324,14 +324,22 @@ export class Browser {
         if (this.config.verboseLogging) {
           this.log.debug('Waiting for dashboard/panel to load', 'timeout', `${options.timeout}s`);
         }
+
         return page.waitForFunction(
-          () => {
+          (isFullPage) => {
+            if (isFullPage) {
```

Comment:
should we add a comment explaining why full page screenshots require a different waiting logic?